### DISCUSSION TITLE
mega2560 won't exit bootloader mode

### DISCUSSION THREAD

**Discussion by kimgalibert** [shanteacontrols/OpenDeck#272] [Q&A]
the board entered bootloader mode via the web configurator.  the sysex update was successful.

but i can't get the board to revert back to normal mode.  

i've tried cycling power, the reset switch, and shorting the gnd and miso pins.

**Comment by paradajz** [comment]
The newer firmware for Mega doesn't support the bootloader anymore. You need to reflash both MCUs with a programmer.

**Comment by kimgalibert** [comment]
i was able to reflash.  had problems with avrdude  error 1507 programmer type linuxspi not found / ubuntu 22.04.

resolved that by using the instructions here to reflash using windows, which was successful for both the mega16u2 and the mega2560.  
https://github.com/shanteacontrols/OpenDeck/discussions/152

but the board still brings up the firmware update screen on the configurator.  it has two solid yellow led's and one solid green.  it accepts and completes the update there, but still won't leave firmware update mode.

**Reply by paradajz** [reply]
You are doing something wrong. As I said, there is no bootloader anymore for this board. Which binaries are you using?

**Reply by kimgalibert** [reply]
i have the 7.7.0 versions of mega2560.bin and mega16u2.bin.  these were downloaded today.

on ubuntu i was not able to flash these.  i am using an arduino leonardo.  it does work under windows.

both the 16u2 and 2560 rec'd 'success' messages under windows.

these are the steps which i followed:  all three completed successfully for both the 2560 and the 16u2.
./avrdude -p atmega16u2 -C ..\etc\avrdude.conf -P usb -c avrispmkII -e -V -u -U lock:w:0xFF:m -U efuse:w:0xF8:m -U hfuse:w:0xD0:m -U lfuse:w:0xFF:m
./avrdude -p atmega16u2 -C ..\etc\avrdude.conf  -P usb -c avrispmkII -U flash:w:mega16u2.hex
./avrdude -p atmega16u2 -C ..\etc\avrdude.conf -P usb -c avrispmkII -V -u -U lock:w:0xEF:m

(not pasting in the similar commands for the 2560)

(edit to add)
also i apologize, it is not 'bootloader' mode, it is 'firmware update' mode that the board is stuck in.

if i connect it and start the configurator it goes to the firmware upload screen.

**Comment by paradajz** [comment]
I will check later.

**Comment by paradajz** [comment]
I have flashed the latest firmware onto my Arduino Mega board and everything works. See the updated guide.
https://github.com/shanteacontrols/OpenDeck/wiki/Flashing-AVR-boards

**Comment by kimgalibert** [comment]
i followed the new procedure and got an avrdude error (board signature 0000) when flashing the mega16u2.

on a lark, i tried to flash the mega2560, that worked.  

unfortunately i still have the situation where the config software moves into firmware update mode as soon as the board is connected.  

shall i assume the 16u2 is fried and move on?

**Comment by kimgalibert** [comment]
i purchased 2 mega2560's but they do not have the second set of icsp pins.  these boards have a CH340 instead of the 16u2.  are these compatible with opendeck?

**Reply by paradajz** [reply]
No they are not.
https://github.com/shanteacontrols/OpenDeck/wiki/Flashing-AVR-boards#arduino-mega2560-1

**Comment by paradajz** [comment]
> shall i assume the 16u2 is fried and move on?

Possibly. Not sure. As I said, all the Megas I have work just fine.

**Comment by kimgalibert** [comment]
it seems likely that the cause of this issue is my attempt to use the opendeck configurator to update the firmware on the mega2560.

prior to that, the board was installed in a guitar and operational for more than a year.

perhaps the configurator should not display an 'update firmware' option for boards which are not supported?  or maybe just a warning listing the boards which should not be updated with the configurator.

**Comment by kimgalibert** [comment]
i just rec'd two 'mega2560 pro minis', on the back they are marked 'www.epalsite.com'.

both flashed fine using the docker procedure.  

they were about $21 each including shipping.
